fava/static/javascript/editor.js

```diff
@@ -254,6 +255,26 @@ export default function initSourceEditor(name) {
       });
     }
   });
+
+  const leaveMessage = 'There are unsaved changes. Are you sure you want to leave?';
+
+  router.on('navigate', () => {
+    if (!editor.getDoc().isClean()) {
+      const leave = confirm(leaveMessage); // eslint-disable-line no-restricted-globals,no-alert
+      if (leave) {
+        editor.getDoc().markClean();
```

Comment:
This seems wrong

Reply:
When navigating away from the (dirty) editor, and then navigating back, the change made before will be gone, but the editor is still marked dirty.